aspnetcore/test/integration-tests.md

```diff
@@ -515,7 +515,7 @@ By default, non-essential cookies aren't preserved across requests when the [GDP
 
 Web host configuration can be created independently of the test classes by inheriting from `WebApplicationFactory` to create one or more custom factories:
 
-1. Inherit from `WebApplicationFactory` and override [ConfigureWebHost](/dotnet/api/microsoft.aspnetcore.mvc.testing.webapplicationfactory-1.configurewebhost). The [IWebHostBuilder](/dotnet/api/microsoft.aspnetcore.hosting.iwebhostbuilder) allows the configuration of the service collection with [ConfigureServices](/dotnet/api/microsoft.aspnetcore.hosting.istartup.configureservices):
+1. Inherit from `WebApplicationFactory` and override [ConfigureWebHost](/dotnet/api/microsoft.aspnetcore.mvc.testing.webapplicationfactory-1.configurewebhost). The [IWebHostBuilder](/dotnet/api/microsoft.aspnetcore.hosting.iwebhostbuilder) allows the configuration of the service collection with [ConfigureServices](/dotnet/api/microsoft.aspnetcore.hosting.istartup.configureservices) which is executed before the app's `Startup.ConfigureServices`. The [IWebHostBuilder](/dotnet/api/microsoft.aspnetcore.hosting.iwebhostbuilder) allows overriding and modiying registered services in the service collection with [ConfigureTestServices](/dotnet/api/microsoft.aspnetcore.testhost.webhostbuilderextensions.configuretestservices):
 
```

Comment:
I guess this is correct considering the example and context currently uses `Web Host` rather than `Generic Host`?

Reply:
I can see the sample at [CustomWebApplicationFactory](https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/test/integration-tests/samples/3.x/IntegrationTestsSample/tests/RazorPagesProject.Tests/CustomWebApplicationFactory.cs) uses `Generic Host` builder in the SUT, but `ConfigureWebHost` in the test. Is that correct as I can see the `ConfigureHost(IHostBuilder builder)` virtual method in `WebApplicationFactory` which I would assume is for `Generic Host`

Reply:
Yes, this only appears in the 2.x part of the doc, so this should be ok.

Reply:
... and the sample code below this is 2.x.

If you're saying that the 2.x code is using `ConfigureHost` (or vice-versa the 3.x code is using `ConfigureWebHost`), then there might be a problem here. AFAIK, this is all ok given that ...

* IIRC, we have proper sign-offs from engineering on this content.
* There haven't been any comments coming in that the code examples/text are wrong between 2.x and 3.x.

I'm 🏃😅 right now on high priority issues, so I'd prefer not to look unless you think there is a problem. If so, let me know ... but I'll need to schedule this for next week.

Reply:
Sorry for my ignorance and this stupid question, but could you expand on "this only appear in the 2.x part of the doc"?

Reason I am asking is because I can see `ConfigureWebHost` in the 3.x samples, and in the doc @docs.microsoft.com I have ASP.NET Core 3.1 selected.

I think I am missing something.

Either way, if you say it's ok I'll take your word for it 👍😃

Reply:
The way these docs work with the version selector on the live site is that there are these markdown lines that set which text is shown. You can see them sprinkled around in the docs ...

```
::: moniker range=">= aspnetcore-3.0"

3.x content here

::: moniker-end

::: moniker range="< aspnetcore-3.0"

2.x content here

::: moniker-end
```

When a reader selects the version on the live site, they see the text in the matching version block ("moniker range").

... and sample cross-links in there (usually) have a `2.x` segment for 2.x and `3.x` for 3.x.

As long as the code is right for that version of the text, it should be ok. We occasionally goof up and show 3.x code in 2.x text or vice-versa, but the community is pretty quick on issuing a good 💀 *death threat* 💀 if we don't fix it. :smile:

Reply:
Awesome thanks for the clarification! Is it worth adding this info to the Contribution Guidelines linked form the root README? (I am assuming the same syntax/logic is used across all docs in this repo?

Reply:
It applies to all of Microsoft docs everywhere. I'm not sure how they've decided to handle it company-wide/.NET Foundation-wide.

I've asked about adding *a lot* of content to the local guidelines (e.g., https://github.com/dotnet/AspNetCore.Docs/issues/2617), but there hasn't been time and/or consensus among the team to do so.